src/System Application/App/AI/src/Azure OpenAI/AzureOpenAIImpl.Codeunit.al

```diff
@@ -480,4 +492,17 @@ codeunit 7772 "Azure OpenAI Impl"
         end;
     end;
 
+    [NonDebuggable]
+    procedure ApproximateTokenCount(Input: Text): Decimal
+    var
+        AverageWordsPerToken: Decimal;
+        TokenCount: Integer;
+        WordsInInput: Integer;
+    begin
+        AverageWordsPerToken := 0.6; // Based on OpenAI estimate
+        WordsInInput := Input.Split(' ', ',', '.', '!', '?', ';', ':', '/n').Count;
```

Comment:
Should we also add '-', '_', '/', '\' etc.? Or maybe we can give a comment on why only these. Often a - combines two words.
Also, we should probably remove all empty results, like "test, abc" I assume will give the result "test", " " (between ',' and ' ') and "abc"

Reply:
We certainly can but this as the name specifies, is just an approximate. If we want a more definite way which covers everything, we should look into https://github.com/openai/tiktoken at a future date. We can create a new function `TokenCount` when we use that.